03/app.js

```diff
@@ -2,38 +2,32 @@ document.addEventListener('DOMContentLoaded', init);
 
 function init() {
     const input = document.querySelector('input');
-    const imagesList = document.querySelector('.images-list');
-    input.addEventListener('change', showFiles);
+    input.addEventListener('change', showFiles);    
+}
 
-    function showFiles(e) {
-        const files = e.target.files;
+function showFiles(e) {
+    const files = [...e.target.files];
         
-        for (let i = 0; i < files.length; i++) {
-            
-            if(files[i].type.includes('image')) {
+    files.forEach(file => {
+        file && file.type.includes('image') ? readFile(file) : alert('Your file is not an image.');
+    });
+}
 
-            const imagesListItemPrototype = document.querySelector('.images-list__item--prototype');
-            const cloneImagesListItemPrototype = imagesListItemPrototype.cloneNode(true);
-    
-            const img = cloneImagesListItemPrototype.querySelector('img');
-            const header = cloneImagesListItemPrototype.querySelector('header');
-            const footer = cloneImagesListItemPrototype.querySelector('footer');            
-                
-            const reader = new FileReader();
-            reader.addEventListener('load', function(e) {
-                header.innerText = files[i].name;
-                footer.innerText = (files[i].size / (1024*1024)).toFixed(2) + 'MB';
-                cloneImagesListItemPrototype.classList.remove('images-list__item--prototype');
-                imagesList.appendChild(cloneImagesListItemPrototype);
-                    
-                imgSrc = e.target.result;
-                img.src = imgSrc;                
-            });
+function readFile(file) {
+    const reader = new FileReader();
+    reader.addEventListener('load', function (e) {    
+        const imagesList = document.querySelector('.images-list');
+        const imagesListItemPrototype = document.querySelector('.images-list__item--prototype');
+        const cloneImagesListItemPrototype = imagesListItemPrototype.cloneNode(true);
```

Comment:
Jeśli masz problem z "gubieniem" zmiennych to wystarczy przekazać je przez parametr.
Rozumiem, że problemem teraz jest to, że nie ma za bardzo jak przekazać tego parametru tak?
Najprościej w tedy uruchomić  funkcję w funkcji tj.
```
function readFile(file) {
    const reader = new FileReader();
    reader.addEventListener('load', function (e) {   
        doSomething(e, file, otherParam)
    }
```

Można też wykorzystać bind(), który tworzy kopie innej funkcji i wtedy możemy przekazać odpowiednie parametry (to już bardziej zaawansowane).
```
function readFile(file) {
    const reader = new FileReader();
    reader.addEventListener('load', doSomething.bind(null, e, file, otherParam))
```

PS. Pierwszy parametr to informacja na co wskazuje `this` - dla nas bez znaczenia dlatego jest `null`